hnn_core/network.py

```diff
@@ -1253,23 +1253,77 @@ def add_connection(self, src_gids, target_gids, loc, receptor,
 
         self.connectivity.append(deepcopy(conn))
 
-    def clear_connectivity(self):
-        """Remove all connections defined in Network.connectivity
+    def _get_expected_connectivities(self, src_types='all'):
+        """Return expected connectivities left after clearng connections.
+
+        Parameters
+        ----------
+        src_types : list | all
+            Connection source types to be cleared
+
+        Returns
+        -------
+        int
+            Number of connections left after the deletion
+            operation
         """
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] in self.external_drives.keys():
-                connectivity.append(conn)
-        self.connectivity = connectivity
-
-    def clear_drives(self):
-        """Remove all drives defined in Network.connectivity"""
-        connectivity = list()
-        for conn in self.connectivity:
-            if conn['src_type'] not in self.external_drives.keys():
-                connectivity.append(conn)
-        self.external_drives = dict()
-        self.connectivity = connectivity
+        if src_types == 'all':
+            return 0
+        deleted_connectivities = 0
+        for src_type in src_types:
+            deleted_connectivities += len(pick_connection(self,
+                                          src_gids=src_type))
+        return len(self.connectivity) - deleted_connectivities
+
+    def clear_connectivity(self, src_types="all"):
```

Comment:
This is a bit of a nitpick. But for stylistic consistency, I suggest using single quotes everywhere unless you need to escape certain characters. You can also check what the PEP 8 document says.

Reply:
https://peps.python.org/pep-0008/#string-quotes

Ok, it makes no recommendation but I personally prefer single quotes :)

Reply:
Sure will use single quotes